MiniTwit.Web.App/Controllers/ApiController.cs

```diff
@@ -34,233 +35,215 @@ IUserRepository userRepository
         {
-            return Json(new {latest = _latest});
-        }
-
-        private void UpdateLatest(int? latestMessage)
-        {
-            if (latestMessage != null)
-            {
-                _latest = latestMessage.Value;
-            }
+            return Json(new GetLatestDTO(_latest));
         }
-        
 
         [Route("[controller]/msgs/")]
         [HttpGet]
         [AllowAnonymous]
-        public async Task<IActionResult> GetMessages([FromQuery(Name = "latest")] int? latestMessage,
-            [FromQuery(Name = "no")] int number = 100)
+        public async Task<IActionResult> GetMessages(
+            [FromQuery(Name = "latest")] int? latestMessage,
+            [FromQuery(Name = "no")] int number = 100
+        )
         {
             UpdateLatest(latestMessage);
-            if (!RequestFromSimulator())
+            if (!IsRequestFromSimulator())
             {
                 return NotAuthorizedError();
             }
             var messages = (await _messageRepository.ReadManyAsync(number))
-                .Select(m => new {content = m.Text, pub_date = m.PubDate, user = m.Author.UserName});
-            
+                .Select(GetMessageDTO.FromMessage);
             return Json(messages);
         }
 
         [Route("[controller]/msgs/{username}")]
         [HttpGet]
         [AllowAnonymous]
-        public async Task<IActionResult> UserGetMessages(string username,
-            [FromQuery(Name = "latest")] int? latestMessage, [FromQuery(Name = "no")] int number = 100)
+        public async Task<IActionResult> UserGetMessages(
+            string username,
+            [FromQuery(Name = "latest")] int? latestMessage,
+            [FromQuery(Name = "no")] int number = 100
+        )
         {
             UpdateLatest(latestMessage);
-            if (!RequestFromSimulator())
+            if (!IsRequestFromSimulator())
             {
                 return NotAuthorizedError();
             }
             var user = await _userRepository.ReadAsyncByUsername(username);
             if (user == null)
             {
-                LogInfo($"Invalid username '{username}' to get messages for");
-                // TODO: This has to be another error, likely 500???
+                LogRequestInfo($"Invalid username '{username}' to get messages for.");
                 return NotFound();
             }    
             var messages = (await _messageRepository.ReadManyFromUserAsync(user.Id,number))
-                .Select(m => new {content = m.Text, pub_date = m.PubDate, user = m.Author.UserName});
+                .Select(GetMessageDTO.FromMessage);
             return Json(messages);
         }
         
 
         [Route("[controller]/msgs/{username}")]
         [HttpPost]
         [AllowAnonymous]
-        public async Task<IActionResult> UserPostMessages(string username,
-            [FromQuery(Name = "latest")] int? latestMessage, [FromBody] MessagePost messagePost)
```

Comment:
At least this does not work. Dont merge until it has run for 2000 messages with the sim locally.

Reply:
I think the follow post action is also broken.